pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -238,6 +238,16 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 }
 
 func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.Server, error) {
+	// try to read from local file (cloud-init) or metadata server first
+	id, err := readInstanceID()
+	if err == nil {
+		server, err := servers.Get(client, id).Extract()
+		if err == nil {
+			return server, nil
```

Comment:
Your comparison to the Rackspace code is incorrect btw.  The rackspace provider still has a `getServerByName` that does exactly the same thing in openstack provider (looks for an ACTIVE server with the indicated name).  The code you linked to in the first comment is a different function that attempts to find the local instance ID, by looking up the local metadata and falling back to using the provided name and `getServerByName()`.  In *this* `probeInstanceID` function, the name is guaranteed to be the local hostname - but this is not the case for the more generic `getServerByName`.